dali/test/python/test_fw_iterators.py

```diff
@@ -879,9 +880,11 @@ def __iter__(self):
 
     def __next__(self):
         batch = []
-        if self.i < self.n:
+        # setting -1 means that no total iteration limmit is set
+        if self.i < self.n and self.total_n != 0:
```

Comment:
Yes -1, and -1 should make this test to always pass. When it is set to >= 0 at the beginning, it will count until it reaches 0 and then will always raise StopIteration. If -1 then this test should always pass.